pkg/kubelet/kubelet.go

```diff
@@ -488,6 +495,41 @@ func NewMainKubelet(
 	return klet, nil
 }
 
+func getPodInfraContainerImagePullSecret(filename string) (*api.Secret, error) {
```

Comment:
That's probably better. I was writing this quickly at the end of the day yesterday and started with it like this, but your suggestion is better.